cpp/examples/protobuf_pointcloud_writer/protobuf_pointcloud_writer.cpp

```diff
@@ -0,0 +1,153 @@
+#define MCAP_IMPLEMENTATION
+
+#include "descriptor.pb.h"
+#include "foxglove/PointCloud.pb.h"
+#include "mcap/writer.hpp"
+#include <chrono>
+#include <cmath>
+#include <fstream>
+#include <iostream>
+#include <random>
+
+#define NS_PER_MS 1000000
+#define NS_PER_S 1000000000
+#define POINTS_PER_CLOUD 1000
+#define FIELDS_PER_POINT 3
+
+// Point represents a point in 3D space.
+struct Point {
+  float x;
+  float y;
+  float z;
+};
+
+// PointGenerator generates random points on a sphere.
+class PointGenerator {
+private:
+  std::mt19937 _generator;
+  std::uniform_real_distribution<float> _distribution;
+
+public:
+  PointGenerator(uint32_t seed = 0)
+      : _generator(seed)
+      , _distribution(0.0, 1.0) {}
+
+  // next produces a random point on the unit sphere, scaled by `scale`.
+  Point next(float scale) {
+    float theta = 2 * M_PI * _distribution(_generator);
+    float phi = acos(1 - 2 * _distribution(_generator));
+    Point point;
+    point.x = float((sin(phi) * cos(theta)) * scale);
+    point.y = float((sin(phi) * sin(theta)) * scale);
+    point.z = float(cos(phi) * scale);
+    return point;
+  }
+};
+
+// WriteLittleEndianFloat writes a little-endian float into the `output` string of bytes at
+// `offset`.
+void WriteLittleEndianFloat(float input, std::string* output, size_t offset) {
+  static_assert(sizeof(uint32_t) == sizeof(float));
+  uint32_t asInt = *reinterpret_cast<uint32_t*>(&input);
```

Comment:
Technically I think this is undefined behavior. You are only allowed to reinterpret_cast arbitrary things to char/uint8: https://stackoverflow.com/a/38744467/23649

Our other mcap code only works on little endian systems anyway, so I think you are safe to continue that trend here:

https://github.com/foxglove/mcap/blob/f341c7f770ee0527a857c5fb780c316ec0d8d521/cpp/mcap/include/mcap/writer.inl#L1057-L1075

I thought we had a static_assert about this somewhere, but maybe we don't because c++17 doesn't provide std::endian 🤦

Reply:
This seems bad - if we want to run our MCAP writing code on ARM-based robots, we surely need to handle this properly.

Reply:
This library works fine on arm32/arm64 platforms. If you can find a real big-endian system that someone might theoretically purchase and use then we can start caring about this.